### PR TITLE
Add eslint and jscs. Plus fix all the lint errors.

### PR DESCRIPTION
**Pull request by Minglee01**
@incuna/js 
please review

Add eslint and fix up the files.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+## Upcoming
+
+* Add file linting using eslint version 18.1.0. https://github.com/sindresorhus/grunt-eslint.
+* Add grunt jscs version 1.2.0.
```
Comment (henrahmagix) [inline CHANGELOG.md]: I don't think the versions or links are needed in a changelog

File: Gruntfile.js [hunk 2]
```diff
@@ -82,6 +89,20 @@ module.exports = function (grunt) {
                 useShortDoctype: true
             }
         },
+        eslint: {
+            all: {
+                options: {
+                    config: 'eslint.json'
```
Comment (henrahmagix) [inline Gruntfile.js]: Should this be a dotfile so it matches other config files? `.eslint.json`

File: Gruntfile.js [hunk 3]
```diff
@@ -120,10 +143,16 @@ module.exports = function (grunt) {
...
+        'eslint',
+        'jscs'
+    ]);
+
     grunt.registerTask('test', [
         'dist',
         'test-module-isolation',
-        'karma:ci'
+        'karma:ci',
+        'lint'
```
Comment (henrahmagix) [inline Gruntfile.js]: We usually lint before doing anything else, so I think this should go as the very first task

File: CHANGELOG.md (hunk 1 above)
Comment (Minglee01) [inline CHANGELOG.md]: its good to have them in. id say.

**Comment by henrahmagix** [comment]
👍

File: Gruntfile.js [hunk 4]
```diff
@@ -1,158 +1,162 @@
-'use strict';
+/* eslint camelcase: off */
+(function () {
+    'use strict';
```
Comment (henrahmagix) [inline Gruntfile.js]: This should be linted as a `node` file, so it doesn't need a wrapping IIFE

File: CHANGELOG.md (hunk 1 above)
Comment (grahamgilchrist) [inline CHANGELOG.md]: yeah it's helpful

File: Gruntfile.js (hunk 2 above)
Comment (grahamgilchrist) [inline Gruntfile.js]: elsewhere I think we've used `.eslintrc`?

**Pull request merged**